bbl_screen-patch/patches/printerui/qml/printer/SpeedAdjust.qml

```diff
@@ -0,0 +1,385 @@
+import QtQuick 2.12
+import QtQuick.Shapes 1.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import UIBase 1.0
+import Printer 1.0
+import X1PlusNative 1.0
+import "../X1Plus.js" as X1Plus
+
+import "qrc:/uibase/qml/widgets"
+import ".."
+
+/* Speed Adjustment UI
+* - Extended print speed adjustment UI that provides greater control over print speed
+* - Gcode is generated by X1Plus.GcodeGenerator.printSpeed() which interpolates
+* Bambu's speed parameters (acceleration, feed rate, and time remaining)
+* - Both Studio and Handy are already compatible, ie they report the correct speed %
+* even when custom speeds are applied
+* - OEM speed selection still uses PrintManager to apply speed profiles - 
+* This feature does not alter OEM speed profiles! 
+* 
+* Important notes:
+* 1) Range is 30% to 180%. Outside of this range, the trendlines are no longer accurate
+* 2) Minimum step size is 2% because Studio and Handy will not display odd-numbered speeds
+* 3) The entire UI for this feature should live inside SpeedAdjust.qml. 
+* 4) This is not a replacement for the OEM speed profiles, and the OEM speed selector
+* is displayed to the user by default.
+*/
+
+Item {
+    id: speedDial
+    width: parent.width
+    height: parent.height
+    anchors.fill: parent
+    property alias speed: dial.value
+    property var targetSpeed:100
+    property var currentSpeed: PrintManager.currentTask.printSpeed
+    property var speedLabel:  (Math.abs(currentSpeed-targetSpeed) > 0) ? `${targetSpeed} / ${currentSpeed}%` : `${currentSpeed}%`
```

Comment:
1) I believe that conditional is usually spelled `currentSpeed != targetSpeed`.
2) I'm sort of thinking that maybe we do not need that anyway?  Like maybe this should always be `${currentSpeed}%`, since it reflects what state the printer is in right now, and the change only takes place when you click "apply"?
```suggestion
    property var speedLabel: `${currentSpeed}%`
```

Reply:
The main reason this exists is because I had initially made a "ramp" feature that would adjust speed at layer changes. Since there is no case now where `targetSpeed` should differ from `currentSpeed`, it's definitely not necessary.

For now I'm saving the ramp feature for later. I think we nearly have a framework that makes `onLayerChange` gcode macros fairly easy to do